freqtrade/exchange/exchange.py

```diff
@@ -1314,8 +1314,8 @@ def extract_cost_curr_rate(self, order: Dict) -> Tuple[float, str, Optional[floa
     # Historic data
 
     def get_historic_ohlcv(self, pair: str, timeframe: str,
-                           since_ms: int, is_new_pair: bool = False,
-                           candle_type: str = '') -> List:
+                           since_ms: int, candle_type: CandleType,
+                           is_new_pair: bool = False) -> List:
```

Comment:
So if we are going record the funding rate a candle, we'll have to do something like 

```
if candle_type == CandleType.FundingRate:
    funding_rates = self._api.fetch_funding_rate_history(pair)
    candles = []
    for rate in funding_rates:
        candles.append([rate['timestamp'], rate['fundingRate'], null, null, null, null])
```

This is what I had in mind when I wrote these comments anyway `:param candle_type: '', mark, index, premiumIndex, or funding_rate`

I saw that you had FUNDINGRATE as a candle value, so I thought I should let you know what my idea was there

Reply:
I do know this - i'm honestly not yet decided on this part - and if you look at the enum, it has a note there saying that funding_rate may not stay there.

Reply:
@samgermain I think i've now seen what needs to be done for funding-rate (as well as to get mark-candles in dry-run mode in a better way).

i'd however prefer to do that in a separate followup PR, as that'll be pretty focused on the exchange class (not all around like this one).

Reply:
Ok, sounds good